production/loki-mixin/alerts.libsonnet

```diff
@@ -70,6 +70,53 @@
               |||, 'cluster', $._config.per_cluster_label),
             },
           },
+          {
+            // Alert if the compactor has not successfully run compaction in the last 24h since the last compaction.
+            alert: 'LokiCompactorHasNotSuccessfullyRunCompaction',
+            expr: |||
+              # The "last successful run" metric is updated even if the compactor owns no tenants,
+              # so this alert correctly doesn't fire if compactor has nothing to do.
+              min (
+                time() - (loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds{} > 0)
+              )
+              by (%s, namespace)
+              > 60 * 60 * 24
```

Comment:
nit: might be worth using a shorter interval here like `3h`? there would be query performance degradation if there are a lot of uncompacted indexes

Reply:
That makes sense yes :)